helm/hpcc/templates/_helpers.tpl

```diff
@@ -1023,23 +1027,71 @@ Pass in dict with root and me
 {{- end -}}
 
 {{/*
-A template to generate Sasha service
-Pass in dict me
+A template to generate the type of a service based on the visibility setting
+Pass in dict with .root, .visibility defined
 */}}
-{{- define "hpcc.addSashaService" }}
-{{- $serviceName := printf "sasha-%s" .me.name }}
+{{- define "hpcc.isVisibilityPublic" }}
+{{- if and (hasKey . "visibility") .visibility -}}
+ {{- if hasKey .root.Values.global "visibilities" -}}
+  {{- if hasKey .root.Values.global.visibilities .visibility -}}
+   {{- $globalServiceInfo := get .root.Values.global.visibilities .visibility -}}
+   {{- if (not (eq $globalServiceInfo.type "ClusterIP")) -}}
+   true
+   {{- end -}}
+  {{- else -}}
+   {{- required (printf "Specified service visibility %s not found in global visibilities section" .service.visibility) nil -}}
```

Comment:
trivia: is required ... nil , preferable to fatal .. ?

Reply:
It seems to give a fairly nice error message. I can try fatal and see if it seems better

Reply:
I can't find any documentation on fatal

Reply:
(though I did find a typo in the error reporting line when I tried to test it so that was handy!)